src/main/resources/Dockerfile

```diff
@@ -8,19 +8,28 @@ USER $TARANTOOL_SERVER_USER:$TARANTOOL_SERVER_GROUP
 RUN groupadd $TARANTOOL_SERVER_GROUP && useradd -m -s /bin/bash $TARANTOOL_SERVER_USER || true
 
 # install dependencies
-RUN yum -y install cmake make gcc gcc-c++ git unzip cartridge-cli && \
+# a yum bug requires setting ulimit, see https://bugzilla.redhat.com/show_bug.cgi?id=1537564
+RUN ulimit -n 1024 && \
+    yum -y install cmake make gcc gcc-c++ git unzip cartridge-cli && \
     yum clean all
 RUN cartridge version
 
-# build
+# build and run
+FROM cartridge-base AS cartridge-app
 ARG CARTRIDGE_SRC_DIR="cartridge"
 ARG TARANTOOL_WORKDIR="/app"
+ARG TARANTOOL_RUNDIR="/tmp/run"
+ARG TARANTOOL_DATADIR="/tmp/data"
+ARG TARANTOOL_LOGDIR="/tmp/log"
+ARG TARANTOOL_INSTANCES_FILE="./instances.yml"
+ARG TARANTOOL_CLUSTER_COOKIE="testapp-cluster-cookie"
+ENV TARANTOOL_WORKDIR=$TARANTOOL_WORKDIR
+ENV TARANTOOL_RUNDIR=$TARANTOOL_RUNDIR
+ENV TARANTOOL_DATADIR=$TARANTOOL_DATADIR
+ENV TARANTOOL_LOGDIR=$TARANTOOL_LOGDIR
+ENV TARANTOOL_INSTANCES_FILE=$TARANTOOL_INSTANCES_FILE
+ENV TARANTOOL_CLUSTER_COOKIE=$TARANTOOL_CLUSTER_COOKIE
 COPY $CARTRIDGE_SRC_DIR $TARANTOOL_WORKDIR
 WORKDIR $TARANTOOL_WORKDIR
-RUN cartridge build --verbose
-
-ENV TARANTOOL_RUNDIR="/tmp/run"
-ENV TARANTOOL_DATADIR="/tmp/data"
-ENV TARANTOOL_INSTANCES_FILE="./instances.yml"
-
-CMD cartridge build && cartridge start --run-dir=$TARANTOOL_RUNDIR --data-dir=$TARANTOOL_DATADIR --cfg=$TARANTOOL_INSTANCES_FILE
+CMD cartridge build && cartridge start --run-dir=$TARANTOOL_RUNDIR --data-dir=$TARANTOOL_DATADIR \
```

Comment:
Isn't it better to install rocks in an image so that the container of this image can be started immediately?

Reply:
I don't completely get what you suggest. The rocks will be installed prior to launching the application according to `CMD` statement. If the contents of `CARTRIDGE_SRC_DIR` are not changed, the container will run almost immediately because the `cartridge build` command will just do nothing and return quickly.

Reply:
@ArtDu  We cat move `cartridge build` step into **cartridge-base** container but we still need to keep it in **cartridge-app** container, because we allow to pass custome value to **CARTRIDGE_SRC_DIR** argument.

Reply:
I realized that my previous code isn't working as I wanted it.

We specify `CARTRIDGE_SRC_DIR` in an image assembling phase. So I wanted to build the application(install rocks) in the assembling phase.
But it doesn't work now, because we use COPY(not ADD) to copy files from the directory to a container. 

In my conception, it would be great if we can collect rocks in an image-building phase. Then check whether nothing changed on the container starting after a container has been built. 

Your changes don't change behavior so that you can merge them.

Reply:
I believe that building the rocks on image building phase is not completely correct since the users can further change the Cartridge app dependencies. That should not lead to rebuilding the image, since only rocks will need to be rebuilt, and this is faster than the whole image rebuild.

Reply:
> I believe that building the rocks on image building phase is not completely correct since the users can further change the Cartridge app dependencies. That should not lead to rebuilding the image, since only rocks will need to be rebuilt, and this is faster than the whole image rebuild.

And that's why I left 2 lines with cartridge build. The first one in RUN command (during image building). The second one in CMD command(on container starting)

Reply:
We can add one more stage, making COPY + RUN a separate layer that will be rebuilt without all the deps.
Also as an idea of future improvement, we may also specify the deps to be installed by `yum` in an external file so that we would be able automatically rebuild the whole image too in case if we want an external dependency for our Cartridge app (like a newer SSL library or whatever).